### ISSUE TITLE
Inconsistency in kwargs of `jax.random.uniform` with `numpy.random.uniform`

### ISSUE THREAD

**Issue by PhilipVinc** [google/jax#4549]
`jax.random.uniform`second argument is `shape`, which *MUST* be an iterable (see the [docs](https://jax.readthedocs.io/en/latest/jax.random.html#jax.random.uniform))

`numpy.random.uniform` accepts `size` which does not need to be an iterable (See the [docs](https://numpy.org/doc/stable/reference/random/generated/numpy.random.uniform.html))

I kind of agree that `shape` is a better keyword for this, but usually you try to reproduce NumPy API, right?

**Comment by mattjj** [comment]
Thanks for the question!

We try to follow the NumPy API with `jax.numpy`, but not with other sub-packages, like `jax.random`. As you know, `jax.random` differs from `numpy.random` in many ways already!

As far as conventions like these go, we tend to err on the side of less polymorphism than NumPy, so in this case I don't think we want to change this. One thing that would change my mind is if we're not being internally consistent within `jax.random`. Another thing is if it was clear this was actually an unambiguous usability loss (where I think enforcing that the `shape` argument always be a Sequence is a usability _win_, though I'm sure reasonable people can disagree!).

WDYT?

**Comment by hawkinsp** [comment]
Closing, given this is as intended. Feel free to reopen to continue the discussion!
